src/module/item/equipment/document.ts

```diff
@@ -13,16 +14,48 @@ class EquipmentPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends
     override prepareBaseData(): void {
         super.prepareBaseData();
 
-        this.system.traits.otherTags ??= [];
+        // Normalize apex data
+        if (this.system.apex) {
+            if (!this.traits.has("apex")) {
+                delete this.system.apex;
+            } else if (!this.isInvested) {
+                this.system.apex.selected = false;
+            }
+        }
+    }
+
+    override prepareSiblingData(this: EquipmentPF2e<ActorPF2e>): void {
+        super.prepareSiblingData();
+
+        // Ensure that there is only one selected apex item
+        if (!this.system.apex) return;
+        const otherApexItems = this.actor.itemTypes.equipment.filter(
+            (e) => e.system.apex && e.isInvested && e !== this
+        );
+        if (this.system.apex.selected) {
+            for (const item of otherApexItems) {
+                item.system.apex!.selected = false;
+            }
+        } else if (
+            !this.system.apex.selected &&
+            this.isInvested &&
+            otherApexItems.every((e) => !e.system.apex!.selected)
+        ) {
+            this.system.apex.selected = true;
+        }
```

Comment:
I think you can simplify some of this by checking for invested earlier. Perhaps you can early return if its not invested?